### PR TITLE
Remove intermediate states copying in Mllama

### PR DESCRIPTION
**Pull request by jkaniecki**
This change removes unnecessary copying of intermediate hidden states at that end of first encoder's transformer forward, by putting hidden states into prelocated tensor.

### CODE REVIEW

**Review by kdamaszk — APPROVED** [review]
LGTM

File: vllm/model_executor/models/mllama.py [hunk 1]
```diff
@@ -509,20 +509,26 @@ def forward(
...
+                encoder_states[hidden_states_idx] = hidden_states
+                hidden_states_idx += 1
             hidden_states = encoder_layer(
                 hidden_states,
                 attention_mask,
             )
 
         if len(self.layers) - 1 in self.output_hidden_states:
-            encoder_states = encoder_states + (hidden_states, )
+            encoder_states[hidden_states_idx] = hidden_states
```
[the remark this answers is not in the document]
Comment (madamczykhabana) [reply vllm/model_executor/models/mllama.py:529]: I guess it all depends whether encoder_layer breaks the graph or if everything runs in a single recipe.

[the remark this answers is not in the document]
Comment (jkaniecki) [reply vllm/model_executor/models/mllama.py:529]: Currently encoder is computed as single recipe - didn't notice any recompilations connected with the change.

[the remark this answers is not in the document]
Comment (jkaniecki) [reply vllm/model_executor/models/mllama.py]: To make sure no recompilation will occure when encoder graph is sliced into smaller ones I changed scalar indexing to tensor indexing

**Comment by jkaniecki** [comment]
Closing this PR as I created one to native vllm : https://github.com/vllm-project/vllm/pull/11295

**Pull request closed without merging**